pyspeckit/spectrum/models/ammonia_radex.py

```diff
@@ -0,0 +1,638 @@
+    # Dipole Moment in cgs (1.476 Debeye)
+    #mu0 = 1.476e-18
+
+    radexdict = grid_interpolator(temperature = tkin, column_density = 1e1**ntot,
+                                 sigma = width, density = 1e1**logdens)
+    numberdict = {'oneone':'11',
+                  'twotwo':'22',
+                  'threethree':'33',
+                  'fourfour':'44'}
+    line_names = ['oneone','twotwo','fourfour']
+    runspec = np.zeros(len(xarr))
+
+    tau_dict = {}
+
+    for linename in line_names:
+        # short variable names for readability
+        frq = freq_dict[linename]
+        aval = aval_dict[linename]
+
+        tau_dict[linename] = radexdict['tau_'+numberdict[linename]]
+
+    components =[]
+    for linename in line_names:
+        voff_lines = np.array(voff_lines_dict[linename])
+        tau_wts = np.array(tau_wts_dict[linename])
+
+        lines = (1-voff_lines/ckms)*freq_dict[linename]/1e9
+        tau_wts = tau_wts / (tau_wts).sum()
+        nuwidth = np.abs(width/ckms*lines)
+        nuoff = xoff_v/ckms*lines
+        tex = radexdict['Tex_'+numberdict[linename]]
+        # tau array
+        tauprof = np.zeros(len(xarr))
+        for kk,nuo in enumerate(nuoff):
+            tauprof += (tau_dict[linename] * tau_wts[kk] *
+                        np.exp(-(xarr.value+nuo-lines[kk])**2 /
+                               (2.0*nuwidth[kk]**2)) * fillingfraction)
+            components.append( tauprof )
+
+        T0 = (h*xarr.value*1e9/kb) # "temperature" of wavelength
+        if tau is not None and thin:
+            #runspec = tauprof+runspec
+            # is there ever a case where you want to ignore the optical depth function? I think no
+            runspec = (T0/(np.exp(T0/tex)-1)-T0/(np.exp(T0/background_tb)-1))*(1-np.exp(-tauprof))+runspec
+        else:
+            runspec = (T0/(np.exp(T0/tex)-1)-T0/(np.exp(T0/background_tb)-1))*(1-np.exp(-tauprof))+runspec
+        if runspec.min() < 0 and background_tb == TCMB:
+            raise ValueError("Model dropped below zero.  That is not possible normally.  Here are the input values: "+
+                    ("tex: %f " % tex) +
+                    ("tkin: %f " % tkin) +
+                    ("ntot: %f " % ntot) +
+                    ("width: %f " % width) +
+                    ("xoff_v: %f " % xoff_v) +
+                    ("fortho: %f " % fortho)
+                    )
+    if verbose or debug:
+        log.info("tkin: %g  tex: %g  ntot: %g  width: %g  xoff_v: %g  fortho: %g  fillingfraction: %g" % (tkin,tex,ntot,width,xoff_v,fortho,fillingfraction))
+
+    if return_components:
+        return (T0/(np.exp(T0/tex)-1)-T0/(np.exp(T0/background_tb)-1))*(1-np.exp(-1*np.array(components)))
+
+    if return_tau:
+        return tau_dict
+
+    return runspec
+
+
+class ammonia_radex_model(model.SpectralModel):
+
+    def __init__(self,npeaks=1,npars=6,
+                 parnames=['tkin','ntot','width','xoff_v','logdens','fortho'],
+                 **kwargs):
```

Comment:
I think this function is identical to the `AmmoniaModel` variant?  If so, `ammonia_radex_model` should just inherit from `ammonia_model` and use `super` to use its `__init__`